README.md

```diff
@@ -2,24 +2,198 @@
+
+* Changing the Standard classification of a current API and/or attribute
+
+* Deprecating an existing API and/or attribute
+
+* Clarifications and corrections to existing Standard language
+
+The Standardization Process remains essentially the same regardless of the type of modification being proposed, although the speed at which a modification progresses through the process may vary considerably based on the magnitude and controversial nature of the proposed change. Discussions regarding a proposal are conducted either on the Issue/PR itself or during the regular community teleconference. Teleconferences provide a high bandwidth discussion format and valuable synchronization mechanism to guide progress. Notes from the teleconferences are archived on the PMIx community's [wiki](https://github.com/pmix/pmix-standard/wiki#meeting-information).
+
+Note that a proposal can be withdrawn by the author(s) at any time for any reason up to the time the proposal is committed to the Standard, assuming acceptance.
+
+#### "Classes" of Standard
+The PMIx Standard is composed of three classes of entries:
+
+* "**Provisional**"" entries in the Standard are officially part of the Standard but are not yet guaranteed to be stable - e.g., their API signature and/or attributes they support may be changed, or the entry can be completely removed if subsequent evaluation proves negative. Changes to provisional entries require warning in at least two minor releases of the Standard document before acceptance - i.e., a proposed change to a provisional entry announced in version A.1 of the PMIx Standard may be committed in version A.3 of the Standard. However, removal of a provisional entry must follow the normal deprecation process described below. Thus, the intent is to provide a relatively easy mechanism by which proposals acceptable in principle to the community can enter the Standard prior to being fully stabilized, thereby allowing early adopters an opportunity to explore the proposed capabilities.
+
+    All proposed additions to the PMIx Standard enter the PMIx Standard under Provisional status. Due to the flexible nature of the Provisional class, no guarantees are made with respect to backward or cross-version compatibility for entries in this class.
+
+* "**Stable**" entries are permanent, non-changing parts of the PMIx Standard subject solely to deprecation. In this class, full guarantees are provided with respect to backward and cross-version compatibility beginning with the version upon which the entry was granted "stable" status. No guarantees are made with respect to versions that included the entry during its "provisional" status. Note that stable APIs may continue to define new supported key-value attribute pairs but may not remove any previous definitions. Thus, the _behavior_ of a stable API may be _extended_ via a new attribute but an existing behavior directed by a current attribute may not be eliminated or modified.
```

Comment:
Just to clarify, does this mean you can only deprecate whole interfaces and cannot deprecate an individual attribute?

Reply:
Hmmm...good point. It does get a little complicated when considering the cross-version issue for a stable definition.

I guess I would propose that we allow deprecation of an individual attribute, but that the deprecation proposal must also include deprecation of the associated behavior for every impacted API. Thus, you don't only deprecate an attribute - you are deprecating all behaviors associated with that attribute.

Make sense?

Reply:
I tried to clarify this - see what you think:

In the class definitions:

> * "**Stable**" entries are permanent, non-changing parts of the PMIx Standard subject solely to deprecation. In this class, full guarantees are provided with respect to backward and cross-version compatibility beginning with the version upon which the entry was granted "stable" status. No guarantees are made with respect to versions that included the entry during its "provisional" status. Note that Stable APIs may continue to define new supported key-value attribute pairs, thereby extending the _behavior_ of a stable API without modifying its signature.

>    Attributes used by APIs in the "Stable" class must also reside in that class. A "Stable" API that wishes to utilize a "Provisional" attribute must therefore first seek elevation of the attribute to the "Stable" class. This is necessary to ensure that all "Stable" parts of the Standard are capable of meeting the compatibility requirements. Provisional" APIs are, of course, free to utilize any "Stable" attribute in addition to anything in the "Provisional" class.

in the elevation to stable status section:

> Note that attributes can be proposed for elevation separate from the APIs that use them. Promotion of one or more attributes independent of any API may, for example, occur when a "Stable" API has expressed a desire to utilize them. In such cases, the proposal should clearly identify the "Stable" APIs making the request and provide some justification for it.

and down in the deprecation section:

> Deprecation of "Stable" attributes can be proposed separately from the APIs that support them - i.e., it is not required that an API which utilizes an attribute be deprecated at the same time as the attribute. The process of deprecating an attribute is, however, somewhat more complicated as it may (a) be used in multiple "Stable" as well as "Provisional" APIs and (b) impact the behavior of an API that supports it. Care must be taken, therefore, to identify _all_ places in the Standard that are impacted by the deprecation proposal. Where "Stable" APIs are impacted, a deprecation warning for the change in _behavior_ must be included along with warning of deprecation of the attribute itself.

Reply:
Thanks @rhc54. Yeah. I think I understand, but just to make sure I'm not misunderstanding the terminology, I'd like to give a concrete example and see if it aligns with what you mean.  If `PMIx_Process_monitor` is marked as "Stable", then all of the attributes (optional or required) that can be passed into `PMIx_Process_monitor` (like `PMIX_MONITOR_HEARTBEAT` and `PMIX_MONITOR_FILE`) must also be "Stable", correct?

What about the scenario where someone wants to extend `PMIx_Process_monitor` to include a new monitoring type besides heartbeat or file, like maybe `PMIX_MONITOR_NETWORK` for example.  Would this new attribute have to be immediately added to PMIx as "Stable" to be usable with `PMIx_Process_monitor`?  Or can it start as "Provisional" and then be moved to "Stable"?

Reply:
You ask a good question that highlights the fact that I left out the attribute "required" vs "optional" explanation. Here is what I added to the "classes" section to help clarify:

At the beginning of that section:

> The PMIx Standard is composed of a combination of APIs and key-value attributes. APIs are intentionally designed to be "lightweight" - i.e., the API signature is generic with a minimal number of fixed parameters in its signature. With few exceptions, APIs are required to include an array of key-value attributes in their signature to facilitate behavioral extensions without modifying the API signature. Thus, PMIx attempts to minimize deprecation and modification of APIs as use models evolve over time.

> Each API can designate a given attribute as either "**required**" or "**optional**" - this indicates whether or not a PMIx implementation of that API is required to support the attribute. PMIx implementations are requested to support optional attributes, but are not required to do so. This is intended to both provide flexibility to PMIx implementors (e.g., to tailor the implementation to their target audience) and to ease adoption of the Standard.

> Attributes can similarly be designated by the caller of a PMIx API as either "required" or "optional". This serves as a directive to the PMIx implementation as to the expected response should the implementation not support that attribute. Attributes provided by the caller that are marked as "required" but are not supported by the API must result in the API returning "not supported". Provided attributes marked as "optional" may be ignored if not supported.

Under "provisional":

> All proposed additions to the PMIx Standard enter the PMIx Standard under Provisional status. Attributes in the "Provisional" class may only be designated as "optional" by an API - i.e., an API definition cannot designate a "Provisional" attribute as "required". Users, however, are always free to "require" that an attribute be supported when calling the API, thus directing the API's response should the attribute not be supported by it.

Under "stable":

> Stable APIs may continue to define and/or adopt new supported key-value attribute pairs, thereby extending the _behavior_ of a stable API without modifying its signature. As previously specified, attributes in the "Provisional" class can only be specified as "optional" - therefore, _required_ attributes used by APIs in the "Stable" class must also reside in that class. A "Stable" API that wishes to utilize a "Provisional" attribute must either first seek elevation of the attribute to the "Stable" class so it can be "required", or must designate that attribute as "optional". This is necessary to ensure that all "Stable" parts of the Standard are capable of meeting the compatibility requirements. "Provisional" APIs are, of course, free to utilize any "Stable" attribute in addition to anything in the "Provisional" class.

Reply:
Ah! That makes a lot of sense.  Thanks @rhc54!